corehq/apps/analytics/tasks.py

```diff
@@ -494,18 +593,51 @@ def track_periodic_data():
         # max number of mobile workers
         submit = []
         for user in users_to_domains:
-            email = user.get('email')
+            email = user.get('email') or user.get('username')
             if not _email_is_valid(email):
                 continue
 
+            email_domain = email.split('@')[-1]
+            if email_domain in blocked_email_domains:
+                metrics_gauge(
+                    'commcare.hubspot_data.rejected.periodic_task.email_domain',
+                    1,
+                    tags={
+                        'email_domain': email_domain,
+                    }
+                )
+                continue
+
+            username_email_domain = user.get('username').split('@')[-1]
+            if username_email_domain in blocked_email_domains:
+                metrics_gauge(
+                    'commcare.hubspot_data.rejected.periodic_task.username',
```

Comment:
To check my understanding: this graph on datadog should always be increasing, typically reporting the same number each night, and then jumping whenever someone adds a new blocked domain, right?

Reply:
that's correct for this metric, yes